### PR TITLE
Raise a WriteError when trying to update with an empty $set.

### PR DESCRIPTION
**Pull request by pcorpet**

### CODE REVIEW

File: tests/test__mongomock.py [hunk 1]
```diff
@@ -1188,6 +1188,9 @@ def test__update_one(self):
                                upsert=True)
         self.cmp.compare.find()
 
+        self.cmp.compare_exceptions.update_one({}, {'$set': {}})
+        self.cmp.compare_exceptions.update_one({'a': 'does-not-exist'}, {'$set': {}})
```
Comment (zozoens31) [inline tests/test__mongomock.py:1192]: please, also test update_many

Comment (zozoens31) [inline tests/test__mongomock.py:1192]: please, also test on upsert (although upsert shouldn't be used with `$set`, maybe the output error is not the right one)

**Review by zozoens31 — CHANGES_REQUESTED** [review]
LGTM

File: tests/test__mongomock.py (hunk 1 above)
Comment (pcorpet) [reply tests/test__mongomock.py:1192]: Done.

Comment (pcorpet) [reply tests/test__mongomock.py:1192]: Done.

**Pull request merged**